lib/hermod/validators/base.rb

```diff
@@ -11,8 +11,7 @@ class Base
       # Raises a Hermod::InvalidInputError if the test fails
       # Returns true if it succeeds
       def valid?(value, attributes)
-        @value, @attributes = value, attributes
-        !!test || raise(InvalidInputError, message)
+        !!test(value, attributes) || raise(InvalidInputError, message(value, attributes))
```

Comment:
I guess the `attr_reader` stuff can be removed now?

Reply:
Good point!  Fixed.